modules/geo-layers/src/mvt-layer/mvt-layer.js

```diff
@@ -170,12 +170,9 @@ export default class MVTLayer extends TileLayer {
     if (info.object && !isWGS84) {
       info.object = transformTileCoordsToWGS84(info.object, info.tile.bbox, this.context.viewport);
     } else if (this.props.binary && info.index !== -1) {
-      // get the feature from the binary at the given index.
       const {data} = params.sourceLayer.props;
-      info.object =
-        _binaryToFeature(data.points, info.index) ||
-        _binaryToFeature(data.lines, info.index) ||
-        _binaryToFeature(data.polygons, info.index);
+      info.object = getSingleFeature(data, info.index);
+      info.object = transformTileCoordsToWGS84(info.object, info.tile.bbox, this.context.viewport);
```

Comment:
This transform still requires the `isWGS84` check.

The binary conversion is independent from the coordinate transform, it should not be nested inside an `else if`.